### PR TITLE
Clean up test structure and add additional tests

### LINKED ISSUE #117: Clean up test suite
### Description
Make sure all tests actually test something worthwhile, and fix folder structure.

### Subtasks
- [x] Folder structure
- [x] Additional tests (~85% cov.)

### PR DESCRIPTION
**Pull request by FredrikAugust**
Changelog
-----------
- Clean up test folder structure
- Add additional tests for services and resolutions

------------------------------------------------------------------------------------
Closes #117

### CODE REVIEW

**Comment by FredrikAugust** [comment]
Hey Lewis? Is this where I get in? 🏎️

File: BackendTests/GraphQL/Resolvers/Helpers/DictionaryStringCollectionStringTypeTest.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        private readonly DictionaryStringCollectionStringType _dictionaryStringCollectionStringType = new();
+
+        private object ResolveField(string field)
+        {
+            return _dictionaryStringCollectionStringType.GetField(field).Resolver.Resolve(new ResolveFieldContext()
+            {
+                Source = new Dictionary<string, Collection<string>>()
+                {
+                    {"key1", new Collection<string>(new List<string> {"value1"})},
+                    {"key2", new Collection<string>(new List<string> {"value2"})},
```
Comment (fredrikbw) [inline BackendTests/GraphQL/Resolvers/Helpers/DictionaryStringCollectionStringTypeTest.cs:22]:
```suggestion
                    {"key2", new Collection<string>(new List<string> {"value2"})}
```

File: BackendTests/GraphQL/Resolvers/Metadata/StoredMetadataTypeTest.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+namespace BackendTests.GraphQL.Resolvers.Metadata
+{
+    public class StoredMetadataTypeTest
+    {
+        private readonly StoredMetadataType _storedMetadataType = new();
+
+        private object ResolveField(string fieldName)
+        {
+            return _storedMetadataType.GetField(fieldName).Resolver.Resolve(new ResolveFieldContext()
```
Comment (fredrikbw) [inline BackendTests/GraphQL/Resolvers/Metadata/StoredMetadataTypeTest.cs:17]:
```suggestion
            return _storedMetadataType.GetField(fieldName).Resolver.Resolve(new ResolveFieldContext
```

**Review by fredrikbw — CHANGES_REQUESTED** [review]
I didn't mark all of them in this review, but you should probably just run a code cleanup to get rid of some redundant declarations, at least on the BackendTests Project. 

Perhaps set up another issue just for running code cleanup on the entire solution (so that this PR doesn't change files that aren't relevant for this PR)

**Comment by FredrikAugust** [comment]
Good call, @fredrikbw. I'll take a look at it.

**Review by Kpaubert — APPROVED** [review]
Looking pretty darn good to me.

**Review by fredrikbw — APPROVED** [review]
Looks G to M

**Pull request merged**